library/agent/Agent.ts

```diff
@@ -609,4 +610,12 @@ export class Agent {
         return this.sendHeartbeatEveryMS;
     }
   }
+
+  public async shutdown() {
+    this.logger.log("Shutting down agent...");
+    if (performance.now() > 30000) {
+      // Only send heartbeat if we are running for more than 30 seconds
+      await this.flushStats(1000);
+    }
```

Comment:
maybe one safety check that we didn't just send a heartbeat? e.g. at least one second?